Foundation/NSURLSession/NSURLSessionTask.swift

```diff
@@ -113,7 +77,11 @@ open class URLSessionTask : NSObject, NSCopying {
         _ = FileManager.default.createFile(atPath: fileName, contents: nil)
         self.tempFileURL = URL(fileURLWithPath: fileName)
         super.init()
-        self.easyHandle = _EasyHandle(delegate: self)
+        if let urlProtocolClass = URLProtocol.getProtocolClass(for: request) {
+            if urlProtocolClass is URLProtocol.Type {
+                self._protocol = (urlProtocolClass as! URLProtocol.Type).init(task: self, cachedResponse: nil, client: nil)
```

Comment:
nit: Can we separate the downcast on a different line of code ? Though this is fine, I'm for better readability!